### PR TITLE
Initial working version

### PR DESCRIPTION
**Pull request by stephen-isc**

### CODE REVIEW

File: src/Vault.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        $this->client_secret = $client_secret;
+        $this->vault = $vault;
+    }
+
+    private function authToken(): string
+    {
+        $cached_token = Cache::get('keyvault_token');
+        if ($cached_token) {
+            return $cached_token;
+        }
```
Comment (danny-isc) [inline src/Vault.php:33]: We should be able to use the following here, which is a little neater:
```
if (Cache::has('keyvault_token')) {
  return Cache::get('keyvault_token');
}
```

**Review by danny-isc — CHANGES_REQUESTED** [review]
As discussed, add squizlabs/phpcodesniffer as a dev dependency, and add the following to the scripts block in composer.json:
"lint:php": "phpcs -n --standard=PSR12 ."

The run it to double check standards.  In future we'll set up github actions to run this & tests

File: src/Vault.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                $response->status()
+            );
+        }
+    }
+
+    public function setVault(string $vault): void
+    {
+        $this->vault = $vault;
+    }
+}
```
Comment (stephen-isc) [inline src/Vault.php:82]: Do you want to be able to restore the vault set in config easily? Was thinking I could set it up so calling without an argument set it back.

Comment (danny-isc) [reply src/Vault.php:82]: I hadn't considered it.  But it's a good idea, so you may as well

Comment (stephen-isc) [reply src/Vault.php:82]: Done.

**Review by danny-isc — APPROVED** [review]

**Pull request merged**